jcl/src/java.base/share/classes/java/lang/Class.java

```diff
@@ -214,6 +214,24 @@
 	private static boolean reflectCacheDebug;
 	private static boolean reflectCacheAppOnly = true;
 
+	/*
+	 * This {@code Dummy} class is created to indicate the cached class value is
+	 * initialized to null ;e.g. {@code cachedDeclaringClass} and {@code cachedEnclosingClass}
```

Comment:
The reason for not able to set `cachedClass` default value to `Dummy.class` is that `j.l.Class` can't be instantiated in `Java`, its instances are created by the VM only, and any instance field with non-null default values have to be set by VM natives.